hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -458,4 +458,27 @@ public static BuildCurveStrategyType fromValue(String value) {
       }
     }
   }
+
+  /**
+   * strategy types for optimize layout for hudi data.
+   */
+  public enum BuildLayoutOptimizationStrategy {
+    ZORDER("z-order"),
+    HILBERT("hilbert");
+    private final String value;
+    BuildLayoutOptimizationStrategy(String value) {
+      this.value = value;
+    }
+
+    public static BuildLayoutOptimizationStrategy fromValue(String value) {
```

Comment:
There's default `valueOf` in every Enum. What do we need this method for?

Reply:
SpaceCurveOptimizeHelper.createOptimizedDataFrameByMapValue will use this method.

Reply:
Why can't it use default `valueOf`?

Reply:
oh sorry,  
 i want use "z-order", "hilbert" as the parmeter not "ZORDER". "HILBERT"
and if we want to call this method in scala.   valueOf will not work.